mantidproject/mantid
# Allow ISIS Powder Gem to deal with sudden spikes in spline that occur in texture mode

**Description of work.**
Changed the cropping of Gem based off the maximum of the spline so that if there is a erroneous peak at the end of the spline, it will not count towards the splines max by ensuring that it only checks for the vanadium splines max within the expected range of a vanadium peak on GEM.
**To test:**
(this test relies on files from the data search archive)

Run the following script with these [files](https://github.com/mantidproject/mantid/files/2539081/test1.zip)
```
from isis_powder import Gem
config_file_path = r"path/to/Gem_config_example.yaml"
Gem_example = Gem(config_file=config_file_path,user_name="test")
Gem_example.create_vanadium(input_mode="Individual",first_cycle_run_no="79218",texture_mode=True, save_all=False)
Gem_example.focus(input_mode="Individual", run_number="79207",texture_mode=True, save_all=False)
```

(remember to update the paths in the config.yaml)
The script should finish without failing due to an array out of bounds error, or to calling crop with a minimum x value greater than the maximum x value in the workspace

Fixes #23913, #23741 



*This does not require release notes* because **Fixes a bug that is only present in the nightly not the release**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
